src/Raven.Client/Documents/Operations/Replication/PullReplicationDefinition.cs

```diff
@@ -1,23 +1,64 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Linq;
 using Raven.Client.Documents.Replication.Messages;
+using Sparrow.Json;
 using Sparrow.Json.Parsing;
 
 namespace Raven.Client.Documents.Operations.Replication
 {
-    public class PullReplicationDefinition : FeatureTaskDefinition
+    public class PullReplicationDefinition : IDynamicJsonValueConvertible
     {
+        [Obsolete("You cannot use Certificates on the PullReplicationDefinition any more, please use the dedicated commands: RegisterReplicationHubAccessOperation and UnregisterReplicationHubAccessOperation")]
+        public Dictionary<string, string> Certificates; // <thumbprint, base64 cert>
+
         public TimeSpan DelayReplicationFor;
+        public bool Disabled;
+
         public string MentorNode;
 
+        public PullReplicationMode Mode = PullReplicationMode.Outgoing;
```

Comment:
This is a public field - is it allowed to modify it and set to `PullReplicationMode.Incoming`?

Reply:
What do you mean, modify it? It is only something that you can manipulate using the commands on the server.
But yes, we expect users that setup incoming replication to set it.

Reply:
I thought that `PullReplicationMode.Outgoing` is the only correct value here because it's assigned by default (why not default it to `None`?). But it can be set to `PullReplicationMode.Incoming`, can't it?
```
new PullReplicationDefinition()
{
  ...
  Mode = PullReplicationMode.Incoming
 ...
}
```

Reply:
The idea is that if you already define PullReplication in your code, nothing changes for you.

Reply:
And yes, you can set to incoming, that is meant to work this way